packages/nextcloud/lib/src/api/webdav/webdav_client.dart

```diff
@@ -571,13 +586,21 @@ class WebDavClient {
     required int contentLength,
     DateTime? lastModified,
     DateTime? created,
+    String? checksum,
   }) {
     if (lastModified != null) {
       request.headers['X-OC-Mtime'] = lastModified.secondsSinceEpoch.toString();
     }
     if (created != null) {
       request.headers['X-OC-CTime'] = created.secondsSinceEpoch.toString();
     }
+    if (checksum != null) {
+      final parts = checksum.split(':');
+      if (parts.length != 2 || !supportedChecksumAlgorithms.contains(parts[0].toLowerCase()) || parts[1].isEmpty) {
```

Comment:
This is hard to read.
Please use a string parsing package like string_scanner (https://pub.dev/packages/string_scanner).

Iirc we already depend on it for the http date parsing.

Reply:
You are right. As this is only for validation a regex would probably be the best option.

Please also document it in the doc comment.